osu.Game/Audio/HitSampleInfo.cs

```diff
@@ -56,6 +69,23 @@ public virtual IEnumerable<string> LookupNames
             }
         }
 
-        public HitSampleInfo Clone() => (HitSampleInfo)MemberwiseClone();
+        /// <summary>
+        /// Creates a new <see cref="HitSampleInfo"/> with overridden values.
+        /// </summary>
+        /// <param name="name">An optional new sample name.</param>
+        /// <param name="bank">An optional new sample bank.</param>
+        /// <param name="suffix">An optional new lookup suffix.</param>
+        /// <param name="volume">An optional new volume.</param>
+        /// <returns>The new <see cref="HitSampleInfo"/>.</returns>
+        public virtual HitSampleInfo With(Optional<string> name = default, Optional<string?> bank = default, Optional<string?> suffix = default, Optional<int> volume = default)
```

Comment:
Nitpicky but I'd maybe change the params to be `newName`, `newBank`, `newSuffix`, `newVolume`, to read better.